Fixed upstream via

commit ca099233b204e9e6d178331a439b50ee01d695d4
Author: cedlemo <cedlemo@gmx.com>
Date:   Fri Dec 25 18:35:13 2015 +0100

    Remove all rb_secure references

and component deprecations/removals.

diff --git a/gnomevfs/src/gnomevfs-directory.c b/gnomevfs/src/gnomevfs-directory.c
index 5bfabfa048253d77..ea9e1188d20a52fd 100644
--- a/gnomevfs/src/gnomevfs-directory.c
+++ b/gnomevfs/src/gnomevfs-directory.c
@@ -105,7 +105,6 @@ directory_make_directory(argc, argv, self)
 							  perm);
 	} else {
 		SafeStringValue(uri);
-		rb_secure(2);
 		result = gnome_vfs_make_directory(RVAL2CSTR(uri), perm);
 	}
 
@@ -121,7 +120,6 @@ directory_remove_directory(self, uri)
 	/* XXX: how to use this?
 	SafeStringValue(uri);
 	*/
-	rb_secure(2);
 	if (RVAL2CBOOL(rb_obj_is_kind_of(uri, g_gvfs_uri))) {
 		result = gnome_vfs_remove_directory_from_uri(
 							RVAL2GVFSURI(uri));
diff --git a/gnomevfs/src/gnomevfs-file.c b/gnomevfs/src/gnomevfs-file.c
index 9cfd6d06eb33d601..efde7816485d16a3 100644
--- a/gnomevfs/src/gnomevfs-file.c
+++ b/gnomevfs/src/gnomevfs-file.c
@@ -185,7 +185,6 @@ file_m_chmod(argc, argv, self)
   GnomeVFSFileInfo *info;
   VALUE value;
 
-  rb_secure(2);
   info = gnome_vfs_file_info_new();
   rb_scan_args(argc, argv, "1*", &r_mode, &paths);
   info->permissions = RVAL2GFLAGS(r_mode, GNOME_VFS_TYPE_VFS_FILE_PERMISSIONS);
@@ -205,7 +204,6 @@ file_m_chown(argc, argv, self)
   GnomeVFSFileInfo *info;
   VALUE value;
 
-  rb_secure(2);
   info = gnome_vfs_file_info_new();
   rb_scan_args(argc, argv, "2*", &r_owner, &r_group, &paths);
   /* XXX: hm...how about -1? works for UNIX, but what else? */
@@ -315,7 +313,6 @@ file_m_truncate(self, uri, length)
 {
   GnomeVFSResult result;
 
-  rb_secure(2);
   /* XXX: how to do this?
    * SafeStringValue(uri);
    */
@@ -843,7 +840,6 @@ static VALUE
 file_truncate(self, length)
      VALUE self, length;
 {
-  rb_secure(2);
   return GVFSRESULT2RVAL(gnome_vfs_truncate_handle(_SELF(self),
                                                    NUM2ULL(length)));
 }
